docs/configure-artifact-repository.md

```diff
@@ -175,33 +175,79 @@ artifacts:
 
 ## Configuring Alibaba Cloud OSS (Object Storage Service)
 
-To configure artifact storage for Alibaba Cloud OSS, please first follow
-the [official documentation](https://www.alibabacloud.com/product/oss) to set up
-an OSS account and bucket.
+Create your bucket and access key for the bucket. Suggest to limit the permission
+for the access key, you will need to create a user with just the permissions you
+want to associate with the access key. Otherwise, you can just create an access key
+using your existing user account.
 
-Once it's set up, you can find endpoint and bucket
-information on your OSS dashboard and then use them like the following to
-configure the artifact storage for your workflow:
+Setup [Alibaba Cloud CLI](https://www.alibabacloud.com/help/en/alibaba-cloud-cli/latest/product-introduction)
+and follow the steps to configure the artifact storage for your workflow:
 
-```yaml
-artifacts:
-  - name: my-art
-    path: /my-artifact
+```bash
+$ export mybucket=bucket-workflow-artifect
+$ export myregion=cn-zhangjiakou
+$ # limit permission to read/write the bucket.
+$ cat > policy.json <<EOF
+{
+    "Version": "1",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+              "oss:PutObject",
+              "oss:GetObject"
+            ],
+            "Resource": "acs:oss:*:*:$mybucket/*"
+        }
+    ]
+}
+EOF
+$ # create bucket.
+$ aliyun oss mb oss://$mybucket --region $myregion
+$ # show endpoint of bucket.
+$ aliyun oss stat oss://$mybucket
+$ #create a ram user to access bucket.
+$ aliyun ram CreateUser --UserName $mybucket-user
+$ # create ram policy with the limit permission.
+$ aliyun ram CreatePolicy --PolicyName $mybucket-policy --PolicyDocument "$(cat policy.json)"
+$ # attch ram policy to the ram user.
+$ aliyun ram AttachPolicyToUser --UserName $mybucket-user --PolicyName $mybucket-policy --PolicyType Custom
+$ # create access key and secret key for the ram user.
+$ aliyun ram CreateAccessKey --UserName $mybucket-user > access-key.json
+$ # create secret in demo namespace, replace demo with your namespace.
+$ kubectl create secret generic $mybucket-credentials -n demo\
+  --from-literal "accessKey=$(cat access-key.json | jq -r .AccessKey.AccessKeyId)" \
+  --from-literal "secretKey=$(cat access-key.json | jq -r .AccessKey.AccessKeySecret)"
```

Comment:
This felt very verbose. I think we can just link to relevant OSS documentation directly?

Reply:
Understand. Let me explain. 
It includes 3 parts, 1) create oss bucket; 2) create ram user and grant oss permission. 3) kubectl to config user into argo. There is not one document link for these steps. We have relevant two aliyun product(oss and ram) and several documents, like oss bucket creation, ram user creation, grant ram user oss permission. It is difficult for aliyun user to start up artifact. 
So I refer AWS S3 configuration to have some cli commands to create oss/ram user, also add two kubectl commands to config argo ConfigMap and secret. It is an end to end steps for user to follow and start up argo workflow artifact.

Reply:
Thank you. Sounds good.